staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -459,31 +459,9 @@ func (cnc *CloudNodeController) syncNode(ctx context.Context, nodeName string) e
 	})
 
 	err = clientretry.RetryOnConflict(UpdateNodeSpecBackoff, func() error {
-		var curNode *v1.Node
-		if cnc.cloud.ProviderName() == "gce" {
-			// TODO(wlan0): Move this logic to the route controller using the node taint instead of condition
-			// Since there are node taints, do we still need this?
-			// This condition marks the node as unusable until routes are initialized in the cloud provider
-			if err := nodeutil.SetNodeCondition(cnc.kubeClient, types.NodeName(nodeName), v1.NodeCondition{
-				Type:               v1.NodeNetworkUnavailable,
-				Status:             v1.ConditionTrue,
-				Reason:             "NoRouteCreated",
```

Comment:
also this looks wrong, since the cluster can be created with ip aliases, and not routes